ocaml/lambda/translcore.ml

```diff
@@ -1651,10 +1666,11 @@ and transl_letop ~scopes loc env let_ ands param case partial warnings =
           transl_ident (of_location ~scopes and_.bop_op_name.loc) env
             and_.bop_op_type and_.bop_op_path and_.bop_op_val Id_value
         in
-        let exp = transl_exp ~scopes and_.bop_exp in
-        let right_layout = layout_exp and_.bop_exp in
+        let exp = transl_exp ~scopes Sort.for_bop_exp and_.bop_exp in
```

Comment:
Since I see this pattern again where we have the type but not the sort, I'm wondering: is there no way to compute a sort from a type?

Reply:
Hopefully the above remarks are clarifying (we can, but want to avoid doing so)